packages/client/src/lib.js

```diff
@@ -30,8 +29,17 @@ const MAX_PUT_RETRIES = 5
 const MAX_CONCURRENT_UPLOADS = 3
 const MAX_CHUNK_SIZE = 1024 * 1024 * 10 // chunk to ~10MB CARs
 
+/** @typedef { import('./lib/interface.js').API } API */
+/** @typedef { import('./lib/interface.js').Service } Service */
+/** @typedef { import('./lib/interface.js').Filelike } Filelike */
+/** @typedef { import('./lib/interface.js').CIDString} CIDString */
+/** @typedef { import('./lib/interface.js').Web3File} Web3File */
+/** @typedef { import('./lib/interface.js').PutOptions} PutOptions */
+/** @typedef { import('./lib/interface.js').UnixFSEntry} UnixFSEntry */
+/** @typedef { import('./lib/interface.js').Web3Response} Web3Response */
```

Comment:
@alanshaw pls to note. The type importing in the client lib has be converted to many simple typedefs, as I think it makes the rest of the JSDoc types through the file more readable.

I was obliged to stop importing the interface as a module as standard did not approve of it, but then had to fix all the tsc warnings and it was not happy with trying to use a namespace like API.Service, so this seemed the least worst, least magic solution to appease both. I could litter the JSDoc comments with these imports, but it felt nicer to have this block that we can ignore, for more readable method comments.